aws/resource_aws_cloudwatch_event_rule.go

```diff
@@ -163,6 +168,7 @@ func resourceAwsCloudWatchEventRuleRead(d *schema.ResourceData, meta interface{}
 	}
 	log.Printf("[DEBUG] Setting boolean state: %t", boolState)
 	d.Set("is_enabled", boolState)
+	d.Set("tags", saveTagsCloudWatchEvents(conn, d, aws.StringValue(out.Arn)))
```

Comment:
This implementation seems off, which likely can be verified with error checking when using `d.Set()`. With aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```go
if err := d.Set("tags", ...); err != nil {
  return fmt.Errorf("error setting tags: %s", err)
}
```

However `saveTagsCloudWatchEvents` is returning an `error` type itself. The value in the second argument for `d.Set()` with a `TypeMap` should be a `map[string]interface{}`. In general, its preferred to keep tag API calls in the Read function (so you can return errors with context) then just set the `tags` attribute like the others.